File: exercises/Exercise052.java
public class Exercise052 {

    public static void main(String[] args) {
        long num;
        int numDigits;
        for (long l = 1; l <= 1_000_000; l++)
        {
            numDigits = Long.toString(l).length();
            for (int i = 2; i <= 6; i++)
            {
                num = l * i;
                if (Long.toString(num).length() != numDigits) break;
                if (!arePermutations(l, num)) break;
                if (i == 6)
                {
                    System.out.println(l);
                    System.exit(0);
                }
            }
        }
    }

    /**
     * This method determines whether two integers are permuations of each other.
     * @param n1 the first number.
     * @param n2 the second number.
     * @return true iff n1 is a permutation of n2.
     */
    public static boolean arePermutations(long n1, long n2)
    {
        int[] digits1 = new int[10];
        int[] digits2 = new int[10];
        for (char c1 : Long.toString(n1).toCharArray())
            digits1[c1 - '0']++;
        for (char c2 : Long.toString(n2).toCharArray())
            digits2[c2 - '0']++;
        for (int digit = 0; digit <= 9; digit++)
        {
            if (digits1[digit] != digits2[digit]) return false;
        }
        return true;
    }
}

File: python/exercise052.py
# This method determines whether two integers are permuations of each other.
# n1 the first number.
# n2 the second number.
# return true iff n1 is a permutation of n2.
def are_permutations(n1, n2):
    digits1 = [0 for i in range(10)]
    digits2 = [0 for i in range(10)]
    for c in str(n1):
        digits1[int(c)] = digits1[int(c)] + 1
    for c in str(n2):
        digits2[int(c)] = digits2[int(c)] + 1
    for digit in range(10):
        if digits1[digit] != digits2[digit]:
            return False
    return True


for i in range(1, 1_000_000):
    for j in range(2, 7):
        num = i * j
        if not are_permutations(i, num):
            break
        if j == 6:
            print(i)
            exit(0)